.github/workflows/conformance-ingress.yaml

```diff
@@ -0,0 +1,153 @@
+name: ConformanceIngress
+
+# Any change in triggers needs to be reflected in the concurrency group.
+on:
+  pull_request:
+    paths-ignore:
+      - 'Documentation/**'
+      - 'test/**'
+  push:
+    branches:
+      - master
+      - ft/master/**
+    paths-ignore:
+      - 'Documentation/**'
+      - 'test/**'
+
+permissions: read-all
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.event.after }}
+  cancel-in-progress: true
+
+env:
+  minikube_version: 1.25.2
+  cilium_cli_version: v0.11.5
+  timeout: 5m
+  echo_server_image: k8s.gcr.io/ingressconformance/echoserver:v0.0.1@sha256:9b34b17f391f87fb2155f01da2f2f90b7a4a5c1110ed84cb5379faa4f570dc52
+
+jobs:
+  ingress-conformance-test:
+    runs-on: ubuntu-latest
+    timeout-minutes: 120
+    steps:
+
+      - name: Install Cilium CLI
+        run: |
+          curl -sSL --remote-name-all https://github.com/cilium/cilium-cli/releases/download/${{ env.cilium_cli_version }}/cilium-linux-amd64.tar.gz{,.sha256sum}
+          sha256sum --check cilium-linux-amd64.tar.gz.sha256sum
+          sudo tar xzvfC cilium-linux-amd64.tar.gz /usr/local/bin
+          rm cilium-linux-amd64.tar.gz{,.sha256sum}
+          cilium version
+
+      - name: Checkout
+        uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
+        with:
+          persist-credentials: false
+
+      - name: Set image tag
+        id: vars
+        run: |
+          if [ ${{ github.event.pull_request }} ]; then
+            SHA=${{ github.event.pull_request.head.sha }}
+          else
+            SHA=${{ github.sha }}
+          fi
+          echo ::set-output name=sha::${SHA}
+
+      - name: Create minikube cluster with multiple nodes
+        run: |
+          # Use at least 3 nodes, so that at least 2 worker nodes are available for scheduling
+          minikube start --network-plugin=cni --cni=false \
+            --docker-opt="default-ulimit=nofile=102400:102400" \
+            --nodes 3 \
+
+      # Start minikube tunnel
+      - name: Setup minikube
+        run: |
+          docker pull ${{ env.echo_server_image }}
+          minikube image load ${{ env.echo_server_image }}
+          minikube tunnel &
+
+      - name: Install Ingress conformance test tool
+        timeout-minutes: 10
+        run: |
+          git clone https://github.com/kubernetes-sigs/ingress-controller-conformance
```

Comment:
Here we are cloning `master` without referencing a specific `ref`, I think we should probably pin a specific SHA just as we do for the rest to mitigate pwn attacks.

Reply:
Actually it got me thinking: maybe instead of doing a manual `git clone` we should do a multi-checkout with `actions/checkout` => https://github.com/actions/checkout#Checkout-multiple-repos-side-by-side

What do you think?

Reply:
it's awesome point :100, I have made the changes with below diff

```diff
diff --git a/.github/workflows/conformance-ingress.yaml b/.github/workflows/conformance-ingress.yaml
index f6af14d988..5929951ae8 100644
--- a/.github/workflows/conformance-ingress.yaml
+++ b/.github/workflows/conformance-ingress.yaml
@@ -71,15 +71,19 @@ jobs:
         env:
           ECHO_SERVER_IMAGE: k8s.gcr.io/ingressconformance/echoserver:v0.0.1@sha256:9b34b17f391f87fb2155f01da2f2f90b7a4a5c1110ed84cb5379faa4f570dc52
 
+      - name: Checkout ingress-controller-conformance
+        uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
+        with:
+          repository: kubernetes-sigs/ingress-controller-conformance
+          path: ingress-controller-conformance
+          ref: cec7cf25b0291933426e256065ef4afc981ea1fa
+          persist-credentials: false
+
       - name: Install Ingress conformance test tool
         timeout-minutes: 10
         run: |
-          git clone https://github.com/kubernetes-sigs/ingress-controller-conformance
           cd ingress-controller-conformance
-          git checkout ${{ env.COMMIT_HASH }}
           make build
-        env:
-          COMMIT_HASH: cec7cf25b0291933426e256065ef4afc981ea1fa
 
       - name: Wait for images to be available
         timeout-minutes: 30

```